x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_view_header_actions.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+
+import { useValues, useActions } from 'kea';
+
+import { EuiPopover, EuiButtonIcon, EuiText, EuiContextMenu, EuiIcon } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { EnginesListLogic } from '../engines/engines_list_logic';
+
+import { EngineViewLogic } from './engine_view_logic';
+
+export const EngineViewHeaderActions: React.FC = () => {
+  const { engineData } = useValues(EngineViewLogic);
+
+  const { openDeleteEngineModal } = useActions(EnginesListLogic);
```

Comment:
should we just add this to the engine view logic instead of re-using the engine list logic? 

I think we can share the modal, but not sure if we should or shouldn't share the logic for a single state value that is essential show or hide modal.

Reply:
oh, but the modal uses the list logic :/ so we'd have to refactor it to take engine as a prop. 

I guess this is fine for now?

Reply:
I think we should refactor the modal and put these into the logic files, I will try to timebox it to see if I can do it today.